_posts/2017-03-05-artsy-technology-stack-2017.md

```diff
@@ -0,0 +1,100 @@
+
+# Artsy Tech Infrastructure
+
+## Front-End + API
+
+In 2015, [dB][db] published our [2015 technology stack][tech2015], in the core Artsy products, though there are a few interesting changes.
+
+What you see today when you go to [artsy.net](https://artsy.net) is a website built with [Ezel.js](http://ezeljs.com), which is a boilerplate for [Backbone](http://backbonejs.org) projects running on [Node](https://nodejs.org) and using [Express](http://expressjs.com) and [Browserify](http://browserify.org). We used to have separate projects for mobile and desktop web, but now they [are merged][force_merge_pr]. It is hosted on [Heroku](http://heroku.com) and uses [Redis](http://redis.io) for caching. Assets, including artwork images, are served from [Amazon S3](http://aws.amazon.com/s3/) via the [CloudFront CDN](http://aws.amazon.com/cloudfront).
+
+What you see today when you open the Artsy iOS app is a mix of Objective-C, Swift and React Native. Objective-C and Swift continue to provide a lot of over-arching cross-View Controller code. While individual representations of Artsy resources tend to be built in React Native. All of our React Native code uses Relay to handle API integration.
+
+Our core-API is a Rails app using [Grape](https://github.com/intridea/grape) which serves JSON. The API [runs on][aws] [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Apache Solr](http://lucene.apache.org/solr), [Elastic Search](https://www.elastic.co) and [Google Custom Search](https://www.google.com/cse). The API service also heavily relies on [Memcached](http://memcached.org).
+
+[confirm with joey about ^ - I know there are search changes]
+
+Most modern code for both the website, and the iOS app use an API meta-layer which is powered by [GraphQL][graphQL]. Our GraphQL server is an [Express](http://expressjs.com) app, using [express-graphql][express-graphql] to provide a single API end-point. The GraphQL API does not access our data directly, but forwards requests to the core API, or to micro-services. We have been migrating front-end display logic into the GraphQL server, to make it easier to build consistent clients.
+
+We continue to have a [public HAL+JSON API](https://developers.artsy.net) for external developers, this API is in active use for a few production services inside Artsy.
+
+## CMS + Writer
+
+We have three major CMS projects:
+
+* One for Partners to upload Show, Fair, Artist and Artwork metadata.
+* One for Editorial, and Partners for writing articles for our magazine.
+* One for in-house Genomers to handle connecting artworks together.
+
+[CMS]
+[Writer]
+[Genome tool]
+
```

Comment:
I would go business by business and highlight services that are specific for the business and services that are generic for everybody to use. Notably after reading this I should be able to pick how our live bidding platform works end-to-end.